src/tickit_devices/eiger/eiger.py

```diff
@@ -55,142 +72,145 @@ def __init__(
+    async def initialize(self) -> None:
+        """Initialize the detector.
+
+        Required for all subsequent operations.
+        """
         self._set_state(State.IDLE)
 
-        json = {"htype": "dseries_end-1.0", "series": "<id>"}
+    async def arm(self) -> None:
+        """Arm the detector.
 
-        LOGGER.debug(json)
+        Required for triggering.
+        """
+        self._series_id += 1
+        self.stream.begin_series(self.settings, self._series_id)
+        self._num_frames_left = self.settings.nimages
+        self._set_state(State.READY)
 
-    async def trigger(self) -> str:
-        """Function to trigger the Eiger.
+    async def disarm(self) -> None:
+        """Disarm the detector.
 
-        If the detector is in an external trigger mode, this is disabled as
-        this software command interface only works for internal triggers.
+        Intended for use when armed. See state diagram in class docstring.
         """
+        self._set_state(State.IDLE)
+        self.stream.end_series(self._series_id)
+
+    async def trigger(self) -> None:
+        """Trigger the detector.
+
+        If the detector is in INTS mode, it will begin acquiring frames the
+        next time update() is called. If it is in EXTS mode, this call will
+        be ignored and acquisition will start based on the parameter to
+        update().
+        INTE and EXTE mode are currently not supported.
+        """
+        LOGGER.info("Trigger requested")
         trigger_mode = self.settings.trigger_mode
-        state = self.status.state
-
-        if state == State.READY and trigger_mode == "ints":
-            self._set_state(State.ACQUIRE)
-
-            for idx in range(0, self.settings.nimages):
-                aquired = Image.create_dummy_image(idx)
-
-                header_json = {
-                    "htype": "dimage-1.0",
-                    "series": "<series id>",
-                    "frame": aquired.index,
-                    "hash": aquired.hash,
-                }
-
-                json2 = {
-                    "htype": "dimage_d-1.0",
-                    "shape": "[x,y,(z)]",
-                    "type": aquired.dtype,
-                    "encoding": aquired.encoding,
-                    "size": len(aquired.data),
-                }
-
-                json3 = {
-                    "htype": "dconfig-1.0",
-                    "start_time": "<start_time>",
-                    "stop_time": "<stop_time>",
-                    "real_time": "<real_time>",
-                }
-
-                LOGGER.debug(header_json)
-                LOGGER.debug(json2)
-                LOGGER.debug(json3)
-
-            return "Aquiring Data from Eiger..."
+
+        if self._is_in_state(State.READY) and trigger_mode == "ints":
+            self._begin_acqusition_mode()
         else:
-            return (
-                f"Ignoring trigger, state={self.status.state},"
+            LOGGER.info(
```

Comment:
So Trigger is ignored if not in a good state to be triggered?

Reply:
Trigger is ignored if not in state READY or not in internal mode, which seems to line up with the behavoir of the real thing